app/src/main/java/com/android/sample/model/recipe/Instruction.kt

```diff
@@ -0,0 +1,25 @@
+package com.android.sample.model.recipe
+
+import com.android.sample.resources.C.TestTag.Instruction.DEFAULT_ICON
+
+/**
+ * Data class representing an instruction in a recipe.
+ *
+ * @property description The description of the instruction.
+ * @property time The time required to complete the instruction in minutes. Nullable.
+ * @property icon The icon representing the instruction, see IconType class to see the different
+ *   icons.
+ */
+data class Instruction(
+    val description: String,
+    val time: String?,
+    val iconType: String?,
+) {
+  private val icon: IconType =
+      if (!iconType.isNullOrEmpty()) IconType(iconType) else IconType(DEFAULT_ICON)
+
+  /** Returns the icon representing the instruction. */
+  fun icon(): IconType {
```

Comment:
Consider renaming icon() to getIcon() to align with naming conventions for getter.
```suggestion
  fun getIcon(): IconType {
```

Reply:
Done